app/javascript/components/Checkout/payment.ts

```diff
@@ -185,6 +189,26 @@ export function computeTipForPrice(state: State, price: number) {
   return Math.round((state.tip.percentage / 100) * price);
 }
 
+function computeTipForFreeProducts(state: State, _price: number, product?: ProductForTip) {
+  if (!product) {
+    const tipAmount = state.tip.type === "fixed" ? (state.tip.amount ?? 0) : 0;
+    return Math.round(tipAmount / state.products.length);
+  }
+
+  const firstProduct = state.products[0];
+  if (!firstProduct) {
+    return 0;
+  }
+  const isFirstProductSelected = firstProduct.permalink === product.permalink;
+
+  if (isFirstProductSelected) {
+    const tipAmount = state.tip.type === "fixed" ? (state.tip.amount ?? 0) : 0;
+    return tipAmount;
+  }
```

Comment:
checks if the computing product is the first product in the cart and then adds the tip to it